### PR TITLE
feat: add Storybook

### PR DESCRIPTION
**Pull request by ngothanhtai**
image

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -37,9 +37,15 @@
     "build:css": "make build-css",
     "test": "react-scripts-ts test --env=jsdom",
     "eject": "react-scripts-ts eject",
-    "commitmsg": "validate-commit-msg"
+    "commitmsg": "validate-commit-msg",
+    "storybook": "start-storybook -p 6006",
+    "build-storybook": "build-storybook"
```
Comment (locphan87) [inline package.json]: `build:storybook` for consistency

File: stories/index.stories.tsx [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import React from 'react';
```
Comment (locphan87) [inline stories/index.stories.tsx]: please add the story next to its component (like test files). no need a dedicated folder for these files

File: tsconfig.json [hunk 3]
```diff
@@ -9,7 +9,7 @@
     "allowJs": true,
     "jsx": "react",
     "moduleResolution": "node",
-    "rootDir": "src",
+    "rootDir": ["src", "stories"],
```
Comment (locphan87) [inline tsconfig.json]: please remove `stories`

**Comment by locphan87** [comment]
we need to lint and format code in .storybook directory as well. can you try to edit `lint-staged` configs and run `yarn precommit`:
```
"lint-staged": {
    "{src,.storybook}/**/*.{ts,tsx}": [
      "prettier --write",
      "yarn lint:fix",
      "git add"
    ]
  }
```

**Comment by ngothanhtai** [comment]
> we need to lint and format code in .storybook directory as well. can you try to edit `lint-staged` configs and run `yarn precommit`:
> 
> ```
> "lint-staged": {
>     "{src,.storybbok}/**/*.{ts,tsx}": [
>       "prettier --write",
>       "yarn lint:fix",
>       "git add"
>     ]
>   }
> ```

Because our storybook file name is `*.storybook.tsx` so lint-staged will lint and format it like other .tsx files so we don't need to do anything

File: .storybook/config.js [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import { configure } from '@storybook/react';
+
+// automatically import all files ending in *.stories.js
+const req = require.context('../src', true, /.storybook.tsx$/);
```
Comment (locphan87) [inline .storybook/config.js]: please use `story` instead of `storybook`

File: .storybook/config.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import { configure } from '@storybook/react';
+
+// automatically import all files ending in *.stories.js
```
Comment (locphan87) [inline .storybook/config.js]: *.stories.js => *.story.tsx

**Pull request merged**